### PR TITLE
Use SCH_USE_STRONG_CRYPTO with SystemDefaults

### PR DESCRIPTION
**Pull request by caesar1995**
The current SslStream implementation was applying SCH_USE_STRONG_CRYPTO only when TLS1.2, TLS1.1 and/or TLS1.0 flags were set. 
In .NET Core 2.0, we introduced a new TLS option (#13075), called 'SystemDefault', which means to use the best (strongest) TLS security 
protocol available in the operating system. But since SystemDefault (equal to all flag bits set to zero) doesn't match any of the TLS bit 
masks, we weren't passing in the SCH_USE_STRONG_CRYPTO flag.

This fix adds a check for SystemDefault and makes sure that the SCH_USE_STRONG_CRYPTO flag is passed to SCHANNEL.
According to the SCHANNEL experts, the current "SystemDefault' will actually use strong crypto settings on current Windows OS's. 
But they advised us that we should always pass in the SCH_USE_STRONG_CRYPTO flag when using 'SystemDefault'.

This is related to internal bug 458042.

### CODE REVIEW

File: src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs [hunk 1]
```diff
@@ -24,6 +24,16 @@ public ClientAsyncAuthenticateTest()
...
         }
 
+        [Fact]
+        public async Task ClientAsyncAuthenticate_SslStreamClientServerNone_UseStrongCryptoSet()
+        {
+            SslProtocols protocol = SslProtocols.None;
+            await ClientAsyncSslHelper(protocol, protocol);
+
+            // Additional manual verification.
+            // Step into the code and verify that the 'UseStrongCrypto' flag is being set.
```
Comment (davidsh) [inline src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: I would change 'UseStrongCrypto' to be 'SCH_USE_STRONG_CRYPTO'.  That is the real name of the SCHANNEL flag and matches the actual source code name `Interop.SspiCli.SCHANNEL_CRED.Flags.SCH_USE_STRONG_CRYPTO`

**Comment by davidsh** [comment]
cc: @dotnet/ncl

File: src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs (hunk 1 above)
Comment (caesar1995) [reply src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: Will fix.

File: src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs [hunk 2]
```diff
@@ -24,6 +24,17 @@ public ClientAsyncAuthenticateTest()
             _log = TestLogging.GetInstance();
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "CI uses old Framework version, which doesn't support SystemDefault")]
```
Comment (davidsh) [inline src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: >"which doesn't support SystemDefault"

I would change this to something like "which doesn't support using SslProtocols.None for SystemDefaultTlsVersions behavior"

Comment (caesar1995) [reply src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: Will change.

File: src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs [hunk 3]
```diff
@@ -24,6 +24,17 @@ public ClientAsyncAuthenticateTest()
             _log = TestLogging.GetInstance();
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "CI uses old Framework version, which doesn't support using SslProtocols.None for SystemDefaultTlsVersions behavior")]
```
Comment (stephentoub) [inline src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: @caesar1995, what version of netfx is required?  Not blocking this PR, but we should a) try to get CI updated to the latest version as soon as possible, and b) track re-enabling these tests when an appropriate version is in play.
cc: @dotnet/dnceng

File: src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs [hunk 4]
```diff
@@ -24,6 +24,17 @@ public ClientAsyncAuthenticateTest()
...
+        public async Task ClientAsyncAuthenticate_SslStreamClientServerNone_UseStrongCryptoSet()
+        {
+            SslProtocols protocol = SslProtocols.None;
+            await ClientAsyncSslHelper(protocol, protocol);
+
+            // Additional manual verification.
+            // Step into the code and verify that the 'SCH_USE_STRONG_CRYPTO' flag is being set.
```
Comment (stephentoub) [inline src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: Is there any way we could automate this?  For example, could we try to connect to a loopback server using SslStream configured to use the weaker crypto and verify that the connection fails when using SslProtocols.None? I don't know if we expose the necessary knobs on SslStream to enable doing that.

File: src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs (hunk 3 above)
Comment (caesar1995) [reply src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: We need at least .NET Framework 4.7 for using SslProtocols.None as SystemDefaultTlsVersions behavior.

File: src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs (hunk 4 above)
Comment (caesar1995) [reply src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs]: SslStream doesn't expose any public APIs for configuring the crypto. From SCHANNEL experts:

>  the current "SystemDefault' will actually use strong crypto settings on current Windows OS's.

But they advised us that we should always pass in the SCH_USE_STRONG_CRYPTO flag when using 'SystemDefault'. There is no behavior difference that can be captured by test for this change.

We could consider adding the scenario you mentioned (SystemDefault with weak crypto) to [SslStreamSystemDefaultsTest.cs](https://github.com/dotnet/corefx/blob/master/src/System.Net.Security/tests/FunctionalTests/SslStreamSystemDefaultsTest.cs) in a different PR, more like verifying SystemDefaultTlsVersions behavior. (if feasible)

**Pull request merged**